### PR TITLE
Home task 4

### PR DESCRIPTION
**Pull request by AlexLeshyk**
Home task 4: Pipes & Directives

### CODE REVIEW

File: src/app/course-page/components/course-item/course-item.component.html [hunk 1]
```diff
@@ -1,13 +1,15 @@
...
     <p>{{ courseItem.description }}</p>
   </div>
   <div class="course-item-buttons">
     <button class="secondary-btn">Edit</button>
     <button class="secondary-btn delete-btn" (click)="delete()">Delete</button>
   </div>
-  <!-- <p>{{counter}}</p> -->
 </div>
+
+<!-- [ngClass]="{fresh: courseItem.dateObj < currentDate && courseItem.dateObj >= freshCourseDate, upcoming: courseItem.dateObj > currentDate }" -->
```
Comment (Lexani) [inline src/app/course-page/components/course-item/course-item.component.html]: Please, remove commented markup

File: src/app/course-page/components/course-page/course-page.component.ts [hunk 2]
```diff
@@ -56,9 +80,13 @@ export class CoursePageComponent implements OnInit {
     console.log('OnInit CoursePage Component', this.counter);
   }
 
-  public onRootDelete(item: CourseItem): void {
+  public onItemDelete(item: CourseItem): void {
     this.courseItems = this.courseItems.filter((course: CourseItem) => course.id !== item.id);
     console.log(item.id);
   }
 
+  onValueInParentChanged(value: string) {
```
Comment (Lexani) [inline src/app/course-page/components/course-page/course-page.component.ts]: I do not recommend using naming with 'Parent' included, please, adjust names.

File: src/app/course-page/components/search-control/search-control.component.html [hunk 3]
```diff
@@ -1,6 +1,6 @@
 <div class="search-block-wrapper">
   <div class="container">
-    <input [(ngModel)]="inputValue">
+    <input type="text" [(ngModel)]="inputValue" (ngModelChange)="onChange($event)"/>
```
Comment (Lexani) [inline src/app/course-page/components/search-control/search-control.component.html]: I suggest remove two-way binding on ngModel and make : 
[ngModel]= "inputValue" and (ngModelChange)="Onchange "  
- and in the function OnChange - you will do: inputValue = $event; modelChangeCallback.emit().

File: src/app/course-page/directives/style.directive.ts [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+export class StyleDirective {
+  @Input('courseItem') courseItem : CourseItem;
+
+  @HostBinding('class.highlighted') highlightedClass = null;
+
+  public currentDate = Date.now();
+  public date = new Date();
+  public freshCourseDate: number = this.date.setDate( this.date.getDate() - 14 );
+
+  constructor( private elRef: ElementRef, private r: Renderer2) {
```
Comment (Lexani) [inline src/app/course-page/directives/style.directive.ts]: Please, do not name renderer as r. Change naming at least to 'renderer'

File: src/app/course-page/directives/style.directive.ts [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    // this.r.setStyle(this.elRef.nativeElement, 'borderColor', 'blue');
+    // this.r.addClass(this.elRef.nativeElement, this.freshCourse);
```
Comment (Lexani) [inline src/app/course-page/directives/style.directive.ts]: remove unused code

File: src/app/course-page/directives/style.directive.ts (hunk 4 above)
Comment (Lexani) [inline src/app/course-page/directives/style.directive.ts]: do not make short names, make understandable names. rename elRef to ElementRef

**Review by Lexani — CHANGES_REQUESTED** [review]

File: src/app/course-page/components/course-item/course-item.component.html (hunk 1 above)
Comment (AlexLeshyk) [reply src/app/course-page/components/course-item/course-item.component.html]: Removed

File: src/app/course-page/directives/style.directive.ts (hunk 5 above)
Comment (AlexLeshyk) [reply src/app/course-page/directives/style.directive.ts]: Removed

File: src/app/course-page/directives/style.directive.ts (hunk 4 above)
Comment (AlexLeshyk) [reply src/app/course-page/directives/style.directive.ts]: Renamed r to renderer

Comment (AlexLeshyk) [reply src/app/course-page/directives/style.directive.ts]: Renamed elRef to ElementRef

File: src/app/course-page/components/course-page/course-page.component.ts (hunk 2 above)
Comment (AlexLeshyk) [reply src/app/course-page/components/course-page/course-page.component.ts]: Renamed names

File: src/app/course-page/components/search-control/search-control.component.html (hunk 3 above)
Comment (AlexLeshyk) [reply src/app/course-page/components/search-control/search-control.component.html]: Removed two-way binding

**Pull request merged**